xarray/core/coordinates.py

```diff
@@ -298,7 +298,7 @@ def __init__(
         else:
             variables = {}
             for name, data in coords.items():
-                var = as_variable(data, name=name)
+                var = as_variable(data, name=name, auto_convert=False)
                 if var.dims == (name,) and indexes is None:
```

Comment:
This place is (I think) the only place where a newly created "dimension coordinate" might not always be an IndexVariable (i.e., when an empty dictionary or any other value than `None` is passed as the `indexes` argument). It would be nice to add a test for this specific case.

The other refactored `as_variable` places in this PR always create an IndexVariable for a dimension coordinate so I think it is covered by the existing tests (invariant checks).

Reply:
I've read this comment many times over, and I don't understand why [this test](https://github.com/pydata/xarray/blob/7c3d2dd2987f855a44d91993f73a9e1aef0a481d/xarray/tests/test_coordinates.py#L26) isn't already testing the case you're asking about (i.e. passing `indexes={}` to the `Coordinates` constructor, hence preventing an `IndexVariable` being automatically created).

Reply:
We might want to add a check in that specific test to make sure that the created "x" variable is not coerced to an `IndexVariable` (before this PR it was still the case even though no `xr.indexes.PandasIndex` was created from it).

Reply:
Okay I added a check both explicitly and for if that warning is raised.